### PR TITLE
Feature/392 - ASC to use AssetRendtions

### PR DESCRIPTION
**Pull request by davidjgonzalez**
Pushed some accidental commits to this branch (on my fork), so did a force push to get the right commit setup - however this force closes previous PRs on said branch .. so this is a dupe of https://github.com/Adobe-Marketing-Cloud/asset-share-commons/pull/409 addressing the comments.

----

This PR introduces the use of AssetRenditions (from previous PRs) to ASC. This includes:

1. Search results Card and List thumbnails
2. Cart list thumbnails
3. Details - Image component
4. Details - Video component

Note this does NOT include **Details - Renditions component** at this time, to retain some continuity between those Renditions and the Download renditions. 

In order to support the Details components, the following has been added:

1. An AssetRenditionsDataSource - which lists the AssetRenditions or selection by Authors, including the ability to filter which AssetRenditions can be displayed per data source. This datasource can be configured to:
   * Exclude renditions by name via `excludeAssetRenditions` property
   * Exclude all renditions provided by a named AssetRenditionDispatcher via `excludeAssetRenditionDispatchers` property
   * Include all renditions in AssetRenditionDispatchers that provide a matching "type" via `allowedAssetRenditionTypes` property. Leaving the type blank on the datasource OR the AssetRenditionDispatcher config will not restrict the listing (ie. if either type is blank, this will not result in filtering any renditions out of the datasource)    

2. A "legacy" dialog JavaScript, that allows the toggling of fieldsets in components, supporting a backward-compatible transition of components to the new paradigm. Note that NEW instances of the affected components will NOT allow them to be configured using legacy parameters; The legacy option ONLY is displayed if the component was previously configured with legacy configurations. This is primarily used internally in the ASC project itself, as customer implementations would/should not use both simultaneously in components they own.

### Search results and Card thumbs

This was a straightforward update and uses the Rendition ComputedProperty to select the `card` or `list` AssetRendition.  

```
<img src="${asset.properties['rendition?name=card']" .... />
```

Resulting in a URL of ... 

```
http://localhost:4502/content/dam/asset-share-commons/en/public/pictures/ruslan-bardash-351288.jpg.renditions/card/asset.rendition
```

#### No legacy config present

image

##### Dialog set to configure (non-Legacy mode) modern Asset Renditions

image

##### Dialog set to use Legacy mode

image

### Dispatcher considerations

Echoing Dispatcher consideration from AssetRendition PR, since the asset rendition URL "ends with" `.../asset.rendition`, ModMimeTypeInfo cannot be used to derive the response content type, and instead, the response headers should be cached by AEM Dispatcher.

### CODE REVIEW

File: core/src/test/java/com/adobe/aem/commons/assetshare/util/UrlUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        assertEquals("/content/dam/test%20folder/ir%c4%81%2bpu%20p%c3%b6%20%26p%c3%aep%e2%98%83.jpeg/_jcr_content/test",
+                UrlUtil.escape("/content/dam/test folder/irā+pu pö &pîp☃.jpeg/jcr:content/test"));
+    }
+
+    @Test
+    public void isEscaped() {
+        assertTrue(UrlUtil.isEscaped("/content/dam/test.png"));
+        assertTrue(UrlUtil.isEscaped("/content/dam/test%20asset.png"));
+        assertTrue(UrlUtil.isEscaped("/content/dam/test%20asset.png/_jcr_content"));
+        assertTrue(UrlUtil.isEscaped("/content/dam/test%20folder/ir%c4%81%2bpu%20p%c3%b6%20%26p%c3%aep%e2%98%83.jpeg/_jcr_content/test"));
```
Comment (bstopp) [inline core/src/test/java/com/adobe/aem/commons/assetshare/util/UrlUtilTest.java]: Need a test case for only `_jcr_content`

Comment (davidjgonzalez) [reply core/src/test/java/com/adobe/aem/commons/assetshare/util/UrlUtilTest.java]: image

Not sure i follow... both test methods use the _jcr_content condition in the 3rd assetXXX test. Code has 100% coverage too.

Comment (bstopp) [reply core/src/test/java/com/adobe/aem/commons/assetshare/util/UrlUtilTest.java]: Should have a test case that does not have any escaped content _except_ the `_jcr_content`:

```
assertTrue(UrlUtil.isEscaped("/content/dam/testasset.png/_jcr_content"));
```

File: core/src/test/java/com/adobe/aem/commons/assetshare/util/UrlUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+        assertFalse(UrlUtil.isEscaped("/content/dam/test asset.png"));
+        assertFalse(UrlUtil.isEscaped("/content/dam/test asset.png/jcr:content"));
```
Comment (bstopp) [inline core/src/test/java/com/adobe/aem/commons/assetshare/util/UrlUtilTest.java]: Same here:

```
assertFalse(UrlUtil.isEscaped("/content/dam/testasset.png/jcr:content"));
```

**Pull request merged**